### ISSUE TITLE
EOFException when trying to compress 32 bit MIPS binaries

### ISSUE THREAD

**Issue by zeuner** [upx/upx#80]
When I try to compress 32 bit MIPS binaries, an EOFException is thrown from PackLinuxElf32::pack2.

I would expect upx to compress the binary, or to provide an explanation if the file format does not allow compression.

## How can we reproduce the issue?

1. Download attached binary.
2. Run "upx simple-mips"

## Please tell us details about your environment.

* UPX version used (`upx --version`):

upx 3.93-git-fdce705e55fb
UCL data compression library 1.03
zlib data compression library 1.2.8
LZMA SDK version 4.43

* Operating System and version:

Linux 4.5.0-rc5

* CPU architecture:

i686

**Comment by zeuner** [comment]
github file uploading doesn't seem to work. Please find the failing binary here: http://quidecco.com/~zeuner/simple-mips

**Comment by jreiser** [comment]
Fixed in 12e57035418acf739c0ccc42ed448bc591c29a7d but note the cleanups d97ff8bb40a6240e0ad707270af53427711d3a02
Thank you for the testcase!
You may find it useful to inspect the output from "ld --verbose" particularly the line such as
```
  PROVIDE (__executable_start = SEGMENT_START("text-segment", 0x400000)); . = SEGMENT_START("text-segment", 0x400000) + SIZEOF_HEADERS;
```
where  SIZEOF_HEADERS helps put Elf32_Ehdr etc inside the first PT_LOAD.
